File: 104.MaximumDepthOfBinaryTree/MarkCSIE/MaximumDepthOfBinaryTree/src/MaximumDepthOfBinaryTree.java

class TreeNode {

    int val;
    TreeNode left;
    TreeNode right;

    TreeNode(int x) {
        val = x;
    }
}

public class MaximumDepthOfBinaryTree {

    public static void main(String[] args) {
        MaximumDepthOfBinaryTree test = new MaximumDepthOfBinaryTree();

        TreeNode root3 = new TreeNode(2);
        root3.left = new TreeNode(3);
        root3.right = new TreeNode(3);
        root3.left.left = new TreeNode(4);
        root3.left.right = new TreeNode(5);
        root3.right.left = new TreeNode(5);
        root3.right.right = new TreeNode(4);
        root3.left.right.left = new TreeNode(8);
        root3.left.right.right = new TreeNode(9);
        root3.right.left.left = new TreeNode(9);
        root3.right.left.right = new TreeNode(8);

        System.out.println(test.maxDepth(root3));
    }

    public int maxDepth(TreeNode root) {
        if (root != null) {
            return Math.max(maxDepth(root.left), maxDepth(root.right)) + 1;
        } else {
            return 0;
        }
    }
}

File: 104.MaximumDepthOfBinaryTree/Ddfish/MaximumDepthOfBinaryTree.swift
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public var val: Int
 *     public var left: TreeNode?
 *     public var right: TreeNode?
 *     public init(_ val: Int) {
 *         self.val = val
 *         self.left = nil
 *         self.right = nil
 *     }
 * }
 */
class Solution {
    func maxDepth(_ root: TreeNode?) -> Int {
        return privateMaxDepth(root,0)
    }
    
    private func privateMaxDepth(_ node : TreeNode?,_ depth : Int) -> Int {
        
        if node == nil {
            return depth
        }
        
        let leftDepth  = privateMaxDepth(node?.left,depth+1)
        let rightDepth = privateMaxDepth(node?.right,depth+1)
        
        return max(leftDepth,rightDepth)
    }
}
